core/src/test/java/org/infinispan/xsite/statetransfer/failures/SiteConsumerTopologyChangeTest.java

```diff
@@ -86,27 +88,29 @@ public BackupReceiverRepository wrap(final CacheContainer wrapOn, final BackupRe
                                    public BackupReceiver getBackupReceiver(String originSiteName, String cacheName) {
                                       return new BackupReceiverDelegator(super.getBackupReceiver(originSiteName, cacheName)) {
                                          @Override
-                                         public void handleStateTransferState(XSiteStatePushCommand cmd) throws Exception {
+                                         public CompletionStage<Void> handleStateTransferState(XSiteStatePushCommand cmd) {
                                             log.debugf("Applying state: %s", cmd);
                                             if (!discard.get()) {
-                                               delegate.handleStateTransferState(cmd);
-                                               return;
+                                               return delegate.handleStateTransferState(cmd);
                                             }
                                             DistributionManager manager = delegate.getCache().getAdvancedCache().getDistributionManager();
                                             synchronized (addressSet) {
                                                //discard the state message when all member has received at least one chunk!
                                                if (addressSet.size() == 3) {
                                                   checkPoint.trigger("before-block");
-                                                  checkPoint.awaitStrict("blocked", 30, TimeUnit.SECONDS);
-                                                  delegate.handleStateTransferState(cmd);
-                                                  return;
+                                                  try {
+                                                     checkPoint.awaitStrict("blocked", 30, TimeUnit.SECONDS);
+                                                  } catch (InterruptedException | TimeoutException e) {
+                                                     return CompletableFutures.completedExceptionFuture(e);
```

Comment:
I think we need a new method in `Checkpoint` throwing only an `org.infinispan.util.concurrent.TimeoutException`, I recently had to write the same code :)